global.c

```diff
@@ -0,0 +1,18 @@
+#include <stdio.h>
+
+int main(void)
+{
+   int units;
+   printf("How many pounds to a firkin of butter?\n");
+   scanf("%u", &units);
+   while(units != 56)
+   {
+      printf("No luck, my friend. Try again.\n");
+      scanf("%d", &units);
+      if(units == 56)
+      {
+         printf("You must have looked it up!\n");
+      }
+   }
```

Comment:
```suggestion
   }
   printf("You must have looked it up!\n");
```

In software we try really hard to avoid repeating ourselves (see https://dzone.com/articles/is-your-code-dry-or-wet). By moving the `printf` to the outside of the `while`, we avoid having two checks that are nearly identical (both comparing against `56`). We also eliminate a bug caused by the duplication (if they guess right on the first try then there's no print!).